### PR TITLE
[Cody GA Docs] Improve and Update Cody Gateway docs

### PR DESCRIPTION
**Pull request by MaedahBatool**
This PR ships improvements to Cody Gateway docs.

## Test plan

Docs added

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@cody-gateway)

### CODE REVIEW

File: doc/cody/core-concepts/cody-gateway.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+>NOTE: You can reach out for more details about Sourcegraph Cody Gateway access available to you and how you can gain access to higher rate limits, quotas, and/or model options.
+
+## Privacy and security
+
+Sourcegraph Cody Gateway does not retain sensitive data (prompt test and source code included in requests, etc.) from any traffic received. Only rate limit consumption per Sourcegraph Enterprise subscription and some high-level diagnostic data (error codes from upstream, numeric/enum request parameters, etc) are tracked.
+
+The code that powers Cody Gateway is also [source-available](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph$+f:cmd/cody-gateway+lang:go&patternType=lucky&sm=1&groupBy=path) for audit.
+
+For more details about Cody Gateway security practices, please reach out to your site administrator. You can also refer to the [Cody usage and privacy notice](https://about.sourcegraph.com/terms/cody-notice) for more privacy details about Cody in general.
```
Comment (eseliger) [inline doc/cody/core-concepts/cody-gateway.md:75]: This used to say account manager which I think is more accurate. Our team is the best to answer their questions, not the customers admin IMO :)

**Review by eseliger — APPROVED** [review]

File: doc/cody/core-concepts/cody-gateway.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Sourcegraph Cody Gateway
+
+<p class="subtitle">Learn how to use Cody Gateway powers the default Sourcegraph provider to facilitate Enterprise instances.</p>
```
Comment (bobheadxi) [inline doc/cody/core-concepts/cody-gateway.md:3]:
```suggestion
<p class="subtitle">Learn how to Cody Gateway powers the default Sourcegraph provider to facilitate Enterprise instances.</p>
```

File: doc/cody/core-concepts/cody-gateway.md (hunk 1 above)
Comment (MaedahBatool) [reply doc/cody/core-concepts/cody-gateway.md:75]: So, will it be your Sourcegraph account manager?

File: doc/cody/core-concepts/cody-gateway.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Sourcegraph Cody Gateway
+
+<p class="subtitle">Learn how Cody Gateway powers the default Sourcegraph provider to facilitate Enterprise instances.</p>
```
Comment (bobheadxi) [inline doc/cody/core-concepts/cody-gateway.md:3]:
```suggestion
<p class="subtitle">Learn how Cody Gateway powers the default Sourcegraph provider for completions and embeddings to facilitate Cody functionality in Enterprise instances.</p>
```

Also, Cody Gateway isn't only available to Enterprise users - it is also used by App OOTB

Comment (bobheadxi) [reply doc/cody/core-concepts/cody-gateway.md:3]: However, covering everything might make this blurb rather long, so maybe we can just keep it vague, e.g.

```suggestion
<p class="subtitle">Learn how Cody Gateway powers the default Sourcegraph provider for completions and embeddings to facilitate Cody features.</p>
```

Comment (MaedahBatool) [reply doc/cody/core-concepts/cody-gateway.md:3]: This is a better version. :)

**Review by bobheadxi — APPROVED** [review]
nice, thanks for taking care of the redirects as well :)

**Pull request merged**